packages/client/src/components/MovieCard/useFavorites.js

```diff
@@ -6,22 +6,40 @@ import { useUserContext } from '../../context/UserContext';
 
 export const useFavorites = (initialFavorites = []) => {
   const [favorites, setFavorites] = useState(initialFavorites);
-  const userId = 1; // placeholder userId
+  const [userId, setUserId] = useState(null);
   const { user } = useUserContext();
   const navigate = useNavigate();
 
+  useEffect(() => {
+    const getUserIdByUid = async () => {
+      if (user && user.uid) {
+        try {
+          const response = await fetch(`${apiURL()}/users/${user.uid}`);
+          const data = await response.json();
+          setUserId(data[0].id);
+        } catch (error) {
+          return error.name === 'Something wrong in the operation';
+        }
+      }
+    };
+
+    getUserIdByUid();
+  }, [user]);
```

Comment:
Hm, it's interesting that userId is not in user context and that there are two separate ids? If I understand correctly :)

Perhaps it would be nice to streamline this within userContext. Otherwise this snippet is obv correct :)